From 229fca435d7fdada25d6943c3c770921ed612a8b Mon Sep 17 00:00:00 2001
From: Erfan Abdi <erfangplus@gmail.com>
Date: Sat, 4 Apr 2026 22:43:10 +0330
Subject: [PATCH] Settings: Remove Vendor Security Patch

Change-Id: I64fddead47599112b5f33b40d4fb07ca1f572eed

Settings: Drop all references to LineageVendorSecurityPatchLevelPreference * Else it crashes about phone -> Android version preference with following logs:

04-13 20:00:26.464 3732 3732 E AndroidRuntime: Caused by: android.content.res.Resources: String resource ID #0x3f09001d 04-13 20:00:26.464 3732 3732 E AndroidRuntime: at android.content.res.Resources.getText(Resources.java:480) 04-13 20:00:26.464 3732 3732 E AndroidRuntime: at android.content.Context.getText(Context.java:1174) 04-13 20:00:26.464 3732 3732 E AndroidRuntime: at com.android.settingslib.metadata.UtilsKt.getPreferenceTitle(Utils.kt:36) 04-13 20:00:26.464 3732 3732 E AndroidRuntime: at com.android.settingslib.preference.PreferenceBinding.bind(PreferenceBinding.kt:98) 04-13 20:00:26.464 3732 3732 E AndroidRuntime: at com.android.settings.deviceinfo.firmwareversion.LineageVendorSecurityPatchLevelPreference.bind(LineageVendorSecurityPatchLevelPreference.kt:42)

Signed-off-by: Volla BuildBot <build@volla.com>
---
 .../firmwareversion/FirmwareVersionScreen.kt  |  1 -
 ...neageVendorSecurityPatchLevelPreference.kt | 76 -------------------
 2 files changed, 77 deletions(-)
 delete mode 100644 src/com/android/settings/deviceinfo/firmwareversion/LineageVendorSecurityPatchLevelPreference.kt

diff --git a/src/com/android/settings/deviceinfo/firmwareversion/FirmwareVersionScreen.kt b/src/com/android/settings/deviceinfo/firmwareversion/FirmwareVersionScreen.kt
index 1af3fb6fcea..5ba1fe5196a 100644
--- a/src/com/android/settings/deviceinfo/firmwareversion/FirmwareVersionScreen.kt
+++ b/src/com/android/settings/deviceinfo/firmwareversion/FirmwareVersionScreen.kt
@@ -72,7 +72,6 @@ open class FirmwareVersionScreen : PreferenceScreenMixin, PreferenceSummaryProvi
             +FirmwareVersionDetailPreference()
             +LineageVersionDetailPreference()
             +SecurityPatchLevelPreference()
-            +LineageVendorSecurityPatchLevelPreference()
             +MainlineModuleVersionPreference()
             +BasebandVersionPreference()
             +KernelVersionPreference()
diff --git a/src/com/android/settings/deviceinfo/firmwareversion/LineageVendorSecurityPatchLevelPreference.kt b/src/com/android/settings/deviceinfo/firmwareversion/LineageVendorSecurityPatchLevelPreference.kt
deleted file mode 100644
index 2781173d7da..00000000000
--- a/src/com/android/settings/deviceinfo/firmwareversion/LineageVendorSecurityPatchLevelPreference.kt
+++ /dev/null
@@ -1,76 +0,0 @@
-/*
- * Copyright (C) 2019-2025 The LineageOS Project
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package com.android.settings.deviceinfo.firmwareversion
-
-import android.content.Context
-import android.os.SystemProperties
-import android.text.format.DateFormat
-import androidx.preference.Preference
-import com.android.settings.R
-import com.android.settingslib.metadata.PreferenceMetadata
-import com.android.settingslib.metadata.PreferenceSummaryProvider
-import com.android.settingslib.preference.PreferenceBinding
-import java.text.ParseException
-import java.text.SimpleDateFormat
-import java.util.Date
-import java.util.Locale
-
-class LineageVendorSecurityPatchLevelPreference :
-    PreferenceMetadata, PreferenceSummaryProvider, PreferenceBinding {
-
-    override val key: String
-        get() = "vendor_security_key"
-
-    override val title: Int
-        get() = org.lineageos.platform.internal.R.string.lineage_vendor_security_patch
-
-    override fun bind(preference: Preference, metadata: PreferenceMetadata) {
-        super.bind(preference, metadata)
-        preference.isSelectable = false
-        preference.isCopyingEnabled = true
-    }
-
-    override fun getSummary(context: Context): CharSequence {
-        var patchLevel = SystemProperties.get(AOSP_VENDOR_SECURITY_PATCH_PROPERTY)
-
-        if (patchLevel.isEmpty()) {
-            patchLevel = SystemProperties.get(LINEAGE_VENDOR_SECURITY_PATCH_PROPERTY)
-        }
-
-        if (patchLevel.isNotEmpty()) {
-            try {
-                val template = SimpleDateFormat("yyyy-MM-dd", Locale.US)
-                val patchLevelDate = template.parse(patchLevel)
-                val format = DateFormat.getBestDateTimePattern(Locale.getDefault(), "dMMMMyyyy")
-                patchLevel = DateFormat.format(format, patchLevelDate).toString()
-            } catch (_: ParseException) {
-                // parsing failed, use raw string
-            }
-        } else {
-            patchLevel = context.getString(R.string.unknown)
-        }
-
-        return patchLevel
-    }
-
-    companion object {
-        const val AOSP_VENDOR_SECURITY_PATCH_PROPERTY: String =
-            "ro.vendor.build.security_patch"
-        const val LINEAGE_VENDOR_SECURITY_PATCH_PROPERTY: String =
-            "ro.lineage.build.vendor_security_patch"
-    }
-}
-- 
2.49.0

